File: my-folder/0011-container-with-most-water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r , res = 0, len(height) - 1, 0

        while l <= r:
            area = min(height[r], height[l])*(r-l)
            res = max(res, area)
            if height[r] > height[l]:
                l += 1
            else:
                r -= 1
        return res

File: my-folder/0011-container-with-most-water/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& h) {
        if(h.size() < 2) return 0;
        int l=0;
        int r=h.size()-1;
        int res = -1;
        while(l<r) {
            int area = min(h[r], h[l])*(r-l);
            res = max(res, area);
            if(h[r] > h[l]) l++;
            else r--;
        }
        return res;
    }
};
